packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -41,29 +43,42 @@ export interface ViewCustomTimings {
 export const THROTTLE_VIEW_UPDATE_PERIOD = 3000
 export const SESSION_KEEP_ALIVE_INTERVAL = 5 * ONE_MINUTE
 
-export function trackViews(location: Location, lifeCycle: LifeCycle) {
+export function trackViews(
+  location: Location,
+  lifeCycle: LifeCycle,
+  onNewLocation: NewLocationListener = () => undefined
+) {
+  onNewLocation = wrapOnNewLocation(onNewLocation)
   const startOrigin = 0
-  const initialView = newView(lifeCycle, location, ViewLoadingType.INITIAL_LOAD, document.referrer, startOrigin)
+  const initialView = newView(
+    lifeCycle,
+    location,
+    ViewLoadingType.INITIAL_LOAD,
+    document.referrer,
+    startOrigin,
+    onNewLocation(location)?.viewName
+  )
   let currentView = initialView
 
   const { stop: stopTimingsTracking } = trackTimings(lifeCycle, (timings) => {
     initialView.updateTimings(timings)
     initialView.scheduleUpdate()
   })
 
-  trackHistory(onLocationChange)
-  trackHash(onLocationChange)
+  const { stop: stopHistoryTracking } = trackHistory(onLocationChange)
+  const { stop: stopHashTracking } = trackHash(onLocationChange)
```

Comment:
was there a need to do that?

Reply:
Yes. When `trackViews` is called, it adds the `onLocationChange` listener to history and hash changes. if we don't remove this listener and call `trackViews` again, we end up with the two listeners being executed on history/hash changes. This caused all the `onNewLocation` callbacks that I added during the tests, to be called from all following tests that triggered a history change. This caused `onNewLocation threw an error ...` to be logged multiple times to the console during tests due to the test case where I threw an error. @BenoitZugmeyer helped me to find the root cause which is that we don't stop history/hash tracking.

Reply:
ok, make sense. 
Thought for future cases: It could be interesting to add this kind of info in the commit description to help the reviewer understand why you are doing the changes